pkg/gotohelm/gochart.go

```diff
@@ -51,14 +51,37 @@ func Load(chartYAML, defaultValuesYAML []byte, render RenderFunc, dependencies .
 }
 
 // LoadValues coheres the provided values into a [helmette.Values] and merges
-// it with the default values of this chart.
+// it with the default values of this chart and all dependencies.
 func (c *GoChart) LoadValues(values any) (helmette.Values, error) {
 	valuesYaml, err := yaml.Marshal(values)
 	if err != nil {
 		return nil, errors.WithStack(err)
 	}
 
-	merged, err := helm.MergeYAMLValues("", c.defaultValues, valuesYaml)
+	merged := map[string]any{}
+	for _, dep := range c.dependencies {
+		depsValues, err := dep.LoadValues(nil)
+		if err != nil {
+			return nil, errors.WithStack(errors.Wrapf(err, "failed to calculate dependency value of %q dependency", dep.metadata.Name))
+		}
+
+		depsValuesByte, err := yaml.Marshal(depsValues)
+		if err != nil {
+			return nil, errors.WithStack(err)
+		}
+
+		merged[dep.metadata.Name], err = helm.MergeYAMLValues("", dep.defaultValues, depsValuesByte)
```

Comment:
Now that this is a single function, I see why it was difficult to follow across two functions. This is already happening in the call to `dep.LoadValues(nil)`, is it not?

If we handled dependencies after the first call to `helm.MergeYAMLValues`, we could pass down `merged.Table(dep.Name)` and avoid having to call `yaml.Marshal` two more times.

Though we'll need to take into consideration `Condition` if we're performing a recursive load within this function.

Alternatively, (I think we discussed this but I don't recall why it wouldn't work) if `LoadValues` only loaded and merged the values for the current chart, we could add: `values[dep.Name] = subdot.Values` to `Dot` and achieve the same functionality.